storage/engine/cgo_flags.go

```diff
@@ -0,0 +1,15 @@
+package engine
+
+import (
+	// Link against the protobuf and rocksdb libaries. This is
+	// explicitly because these Go libraries do not export any Go
+	// symbols.
+	_ "github.com/cockroachdb/c-protobuf"
+	_ "github.com/cockroachdb/c-rocksdb"
+)
+
+// #cgo CXXFLAGS: -std=c++11
+// #cgo CPPFLAGS: -I ../../../c-protobuf/internal/src -I ../../../c-rocksdb/internal/include
+// #cgo darwin LDFLAGS: -Wl,-undefined -Wl,dynamic_lookup
```

Comment:
LDFLAGS are propagated from one package to its dependents; do we need to put this here (as opposed to leaving it in the c-rocksdb and c-protobuf repos)?

Is this really a darwin vs linux issue, or are we using OS as a proxy for GCC vs clang-based toolchains? The limited flexibility of go build flags might be a problem here if we want to support more platforms.